rest_framework/fields.py

```diff
@@ -210,16 +210,6 @@ def bind(self, field_name, parent):
         Called when a field is added to the parent serializer instance.
         """
 
-        # In order to enforce a consistent style, we error if a redundant
-        # 'source' argument has been used. For example:
-        # my_field = serializer.CharField(source='my_field')
-        assert self.source != field_name, (
-            "It is redundant to specify `source='%s'` on field '%s' in "
-            "serializer '%s', because it is the same as the field name. "
-            "Remove the `source` keyword argument." %
-            (field_name, self.__class__.__name__, parent.__class__.__name__)
-        )
-
```

Comment:
We definatly *do* want to keep this check. This style is simply unneccessary and confusing.

This:

    foo = fields.IntegerField(source='foo')

Is absolutely poor style, and we shouldn't allow it.